sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,1126 @@
+        Error error = new Error().setCode(serviceError.getCode()).setMessage(serviceError.getMessage())
+            .setTarget(serviceError.getTarget());
+        return new DetectLanguageResult(errorDocument.getId(), error, true);
+    }
+
+    /**
+     * Helper method to create a {@link DetectLanguageResult} for a valid document.
+     *
+     * @param documentLanguage The valid document.
+     * @return A {@link DetectLanguageResult} equivalent for the document.
+     */
+    private static DetectLanguageResult convertToDetectLanguageResult(final DocumentLanguage documentLanguage) {
+        // TODO (savaity): confirm the primary language support from service
+        return new DetectLanguageResult(documentLanguage.getId(), documentLanguage.getStatistics(),
+            documentLanguage.getDetectedLanguages().get(0), documentLanguage.getDetectedLanguages());
+    }
+
+    // Named Entity
+
+    /**
+     * TODO (shawn): add doc
+     *
+     * @param text the text to be analyzed.
+     * @return A {@link Mono} containing the {@link NamedEntityResult named entity} of the text.
+     * @throws NullPointerException if {@code text} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<NamedEntityResult> recognizeEntities(String text) {
+        return recognizeEntitiesWithResponse(text, null).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * TODO (shawn): add doc
+     *
+     * @param text the text to be analyzed.
+     * @param language TODO (shawn): add doc
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} has the
+     * {@link NamedEntityResult named entity} of the text.
+     * @throws NullPointerException if {@code text} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<NamedEntityResult>> recognizeEntitiesWithResponse(String text, String language) {
+        try {
+            return withContext(context -> recognizeEntitiesWithResponse(text, language, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<NamedEntityResult>> recognizeEntitiesWithResponse(String text, String language, Context context) {
+        List<TextDocumentInput> documentInputs = new ArrayList<>();
+        // TODO (shawn): update/validate inputs and id assigning
+        documentInputs.add(new TextDocumentInput(Integer.toString(0), text, language));
+        return recognizeBatchEntitiesWithResponse(documentInputs, null, context).flatMap(response -> {
+            Iterator<NamedEntityResult> responseItem = response.getValue().iterator();
+            if (responseItem.hasNext()) {
+                return Mono.just(new SimpleResponse<>(response, responseItem.next()));
+            }
+            return monoError(logger, new RuntimeException("Unable to recognize entities for the provided text."));
```

Comment:
Should document in the public JavaDocs that a RuntimeException is thrown. (If that's the correct error?). Same question as the one above... surprised the service doesn't return a 4xx if its not recognized..